### PR TITLE
Update checkpoints and chain work

### PR DESCRIPTION
**Pull request by rnicoll**
Most crucially this means mining actually works.

### CODE REVIEW

**Comment by rnicoll** [comment]
Full sync tests are currently running, this is ready for review but should not be merged until those have completed.

File: src/chainparams.cpp [hunk 1]
```diff
@@ -181,26 +180,24 @@ class CMainParams : public CChainParams {
...
-            ( 128370, uint256S("0x3f9265c94cab7dc3bd6a2ad2fb26c8845cb41cff437e0a75ae006997b4974be6"))
             ( 145000, uint256S("0xcc47cae70d7c5c92828d3214a266331dde59087d4a39071fa76ddfff9b7bde72"))
-            ( 250000, uint256S("0x0e4bcfe8d970979f7e30e2809ab51908d435677998cf759169407824d4f36460"))
-            ( 350000, uint256S("0x2bdcba23a47049e69c4fec4c425462e30f3d21d25223bde0ed36be4ea59a7075"))
             ( 371337, uint256S("0x60323982f9c5ff1b5a954eac9dc1269352835f47c2c5222691d80f0d50dcf053"))
             ( 450000, uint256S("0xd279277f8f846a224d776450aa04da3cf978991a182c6f3075db4c48b173bbd7"))
-            ( 550000, uint256S("0xea8ed5430b221549a6a26f104b424ffd782ff4c8409bbbc5eaf3d83932825691"))
-            ( 650000, uint256S("0x486fcebc9a7288676a7614e1b6fd085d5d71019aead17d354a8bc2c3fde516e9"))
             ( 771275, uint256S("0x1b7d789ed82cbdc640952e7e7a54966c6488a32eaad54fc39dff83f310dbaaed"))
+            ( 1000000, uint256S("6aae55bea74235f0c80bd066349d4440c31f2d0f27d54265ecd484d8c1d11b47"))
```
Comment (langerhans) [inline src/chainparams.cpp]: The other havae a `0x` prefix, I guess it's optional?

File: src/chainparams.cpp [hunk 2]
```diff
@@ -181,26 +180,24 @@ class CMainParams : public CChainParams {
...
+            ( 1750000, uint256S("5c8e7327984f0d6f59447d89d143e5f6eafc524c82ad95d176c5cec082ae2001"))
+            ( 2000000, uint256S("9914f0e82e39bbf21950792e8816620d71b9965bdbbc14e72a95e3ab9618fea8"))
         };
 
         chainTxData = ChainTxData{
-            // Data as of block 00000000000000000166d612d5595e2b1cd88d71d695fc580af64d8da8658c23 (height 446482).
-            1435666139, // * UNIX timestamp of last checkpoint block
-            19567197,   // * total number of transactions between genesis and last checkpoint
+            // Data as of block 893297d89afb7599a3c571ca31a3b80e8353f4cf39872400ad0f57d26c4c5d42 (height 2031142).
+            1514567911, // * UNIX timestamp of last checkpoint block
```
Comment (langerhans) [inline src/chainparams.cpp]: Block 2mil has a timestamp of 1512600918 for me on my 1.10.1 node.

File: src/chainparams.cpp [hunk 3]
```diff
@@ -335,12 +331,14 @@ class CTestNetParams : public CChainParams {
             ( 591117, uint256S("0x5f6b93b2c28cedf32467d900369b8be6700f0649388a7dbfd3ebd4a01b1ffad8"))
             ( 658924, uint256S("0xed6c8324d9a77195ee080f225a0fca6346495e08ded99bcda47a8eea5a8a620b"))
             ( 703635, uint256S("0x839fa54617adcd582d53030a37455c14a87a806f6615aa8213f13e196230ff7f"))
+            ( 1000000, uint256S("1fe4d44ea4d1edb031f52f0d7c635db8190dc871a190654c41d2450086b8ef0e"))
```
Comment (langerhans) [inline src/chainparams.cpp]: Prefix here too.

File: src/chainparams.cpp [hunk 4]
```diff
@@ -335,12 +331,14 @@ class CTestNetParams : public CChainParams {
...
+            ( 1202214, uint256S("0xa2179767a87ee4e95944703976fee63578ec04fa3ac2fc1c9c2c83587d096977"))
         };
 
         chainTxData = ChainTxData{
-            // Data as of block 00000000c2872f8f8a8935c8e3c5862be9038c97d4de2cf37ed496991166928a (height 1063660)
-            1440601451, // * UNIX timestamp of last checkpoint block
-            1119061,    // * total number of transactions between genesis and last checkpoint
+            // Data as of block 73864d715723d1b4c7cf41a71aafd648455bba323e0951fde47626c6b19a3c0b (height 1202227)
+            1514565004, // * UNIX timestamp of last checkpoint block
```
Comment (langerhans) [inline src/chainparams.cpp]: I don't have testnet synced but chain.so lists 1495653305 for the last checkpoint. Or am I just understanding that comment wrong?

Comment (rnicoll) [reply src/chainparams.cpp]: Ah crud I've forgotten to actually put the last checkpoint into, y'know, the checkpoints.

**Pull request merged**